rust/connlib/tunnel/src/sockets.rs

```diff
@@ -254,28 +262,40 @@ impl Socket {
         }
     }
 
-    fn poll_send_ready(&self, cx: &mut Context<'_>) -> Poll<io::Result<()>> {
-        self.socket.poll_send_ready(cx)
+    fn poll_flush(&mut self, cx: &mut Context<'_>) -> Poll<io::Result<()>> {
+        loop {
+            match self.socket.try_io(Interest::WRITABLE, || {
+                self.state
+                    .send((&self.socket).into(), &self.buffered_transmits)
```

Comment:
If the sending is done in `poll_flush`, that means nothing is actually written to the network unless something is always calling `poll_flush`, right?

It took me a while to figure out, normally flushing is optional on sockets. But here if I never flushed, it would queue forever

Reply:
Yes, we flush on every iteration of `Io::poll` though which is pretty frequent.

Essentially, we flush if nothing else can make progress without reading more packets from the device.